lib/puppet/type/exec.rb

```diff
@@ -407,6 +407,7 @@ def check(value)
       # If the file exists, return false (i.e., don't run the command),
       # else return true
       def check(value)
+        debug(_("Checking that '%{creates_path}' exists") % { creates_path: value })
```

Comment:
Minor comment: Might be worth adding a translator note that `%{creates_path}` is a filesystem path, or renaming `creates_path` to make it more apparent to the translators, since they're only going to see the string passed to `_()` without any of the context around it.

Reply:
@jhelwig I'd be happy to. How does one add a translator note?

Reply:
https://github.com/puppetlabs/puppet/blob/ba4d1a1aba0095d3c70b98fea5c67434a4876a61/lib/puppet/interface/option_builder.rb#L44

Should be able to add a comment just before the string to be translated like this one.

Reply:
```
#TRANSLATORS The value being checked is a filesystem path and should not be translated.
debug(_("Checking that filesystem path '%{creates_path}' exists") % { creates_path: value })
```

I feel like we could be bikeshedding on the wording for a while here, but how does the above example look?

In my example above, the debug message will look like:
```
Checking that filesystem path `/path/to/creates` exists
```

Hmm, or to make it clear to the user that this is the `creates` path being chceked, should the message be something like:
```
Checking that creates path `/path/to/creates` exists
```

Reply:
I like the second one better, but I have no idea which would be clearer in translation.  Hmm, might be able to reference the property, similar to:
```
#TRANSLATORS 'creates' is a parameter name and should not be translated
debug(_("Checking that 'creates' path '%{creates_path}' exists") % { creates_path: value })

Upon reflection, I don't think the translators touch inferred value names.  @ehom , can you comment on this last point?